jax/experimental/jax2tf/jax2tf.py

```diff
@@ -206,19 +206,23 @@ def check_arg(a):
                "be NumPy array, scalar, tf.Variable, or tf.Tensor")
         raise TypeError(msg)
     tree_util.tree_map(check_arg, args)
+    tree_util.tree_map(check_arg, list(kwargs.values()))
 
     # Name input tensors
     args = tuple(
         tree_util.tree_map(lambda x, i=i: tf.identity(x, f"jax2tf_arg_{i}"), a)  # type: ignore
         for i, a in enumerate(args))
+    kwargs = {k: tf.identity(v, f"jax2tf_arg_{k}") for k, v in kwargs.items()}
 
     # This function may take pytrees of TfVals. We can only set
     # tf.custom_gradient on functions that take a flat argument list.
-    args_flat, in_tree = tree_util.tree_flatten((args, {}))
+    args_flat, in_tree = tree_util.tree_flatten((args, kwargs))
 
     if polymorphic_shapes is None:
       polymorphic_shapes_ = (None,) * len(args)
     else:
+      if kwargs:
```

Comment:
I think that what may be better to do here is to say that we support polymorphic_shapes only for the positional arguments, and do not raise this error. The check in line 226 should still work, and your changes below in line 236 should take care of setting the kwargs to be shape monomorphic. So, the only thing to do is to drop this check here, and update the docstring.

BTW, this suggestion follows the model of what we do for vmap, where we have an `in_axes` (they apply only to positional arguments).

Reply:
Excellent. That's great.